packages/editor/src/components/provider/index.native.js

```diff
@@ -63,6 +64,16 @@ class NativeEditorProvider extends Component {
 		this.subscriptionParentUpdateHtml = subscribeUpdateHtml( ( payload ) => {
 			this.updateHtmlAction( payload.html );
 		} );
+
+		this.subscriptionParentMediaAppend = subscribeMediaAppend( ( payload ) => {
+			const blockName = 'core/' + payload.mediaType;
+			const newBlock = createBlock( blockName, {
+				id: payload.mediaId,
+				[ payload.mediaType === 'image' ? 'url' : 'src' ]: payload.mediaUrl,
+			} );
+
+			this.props.insertBlock( newBlock, this.props.blockCount );
```

Comment:
The only problem with this is that all images but the first one will be added at the end of the post.

Not sure if at this point we have enough information to add each new image when it corresponds? 🤔 
For sure can be an improvement for the future.